src/controllers/projectCtrl.js

```diff
@@ -37,45 +37,46 @@ exports.deleteProject = (req, res) => {
 };
 
 exports.updateProjectInfos = (req, res) => {
-  Project.findByIdAndUpdate(req.params.projectId, req.body, {
-    new: true,
-  })
+  Project.findByIdAndUpdate(req.params.projectId, req.body, { new: true })
     .then((project) => {
       if (!project)
         res
           .status(400)
-          .json({ message: 'No Item updated: The project does not exist' });
-      else
-        res.status(200).json({
-          message: 'The project has been modified correctly!',
-          data: project,
-        });
+          .json({ message: 'No item updated: project not found!' });
+      else res.status(200).json({ message: 'Project updated!', data: project });
     })
     .catch((error) =>
-      res.status(401).json({ message: 'Invalid Request!', error })
+      res.status(400).json({ message: 'Invalid Request!', error })
     );
 };
 
 exports.addUserToProject = (req, res) => {
-  Project.findByIdAndUpdate(
-    req.params.projectId,
-    {
-      $push: {
-        users: {
-          id: req.body.userId,
-          role: req.body.role,
-        },
-      },
-    },
-    {
-      new: true,
-      upsert: true,
-    }
-  )
+  // check if project exist
+  Project.findById(req.params.projectId)
```

Comment:
Do you try something like this : 
```js
  Project.findOneAndUpdate(
    { id: projectId, 'users.id': { $ne: userId } },
    { $addToSet: { users: { id: new ObjectId(userId) } } },
    { new: true }
  )
    .then((project) => {
      if (!project) {
        return res.status(400).json({
          message: 'User already in this project',
          data: null,
        });
      }
      res.status(200).json({
        message: 'The project has been modified correctly!',
        data: project,
      });
    })
    .catch((error) => {
      console.log(error);
      res.status(401).json({ message: 'Invalid Request!', error });
    });```

With this part of code in projectModel.js file
```
projectSchema.pre('findOneAndUpdate', async function (next) {
  const project = this;

  // Test if we wants to add user in project
  if (this.getQuery()['users.id']) {
    const userId = this.getQuery()['users.id']['$ne'];
    const isUserExist = await userModel.findById(userId);
    if (!isUserExist) next('Unknown user');
  }

  next();
});
```